util/Mongoose.hx

```diff
@@ -312,21 +325,22 @@ class Mongoose {
 		return objDecl;
 	}
 
-	static function classFieldToSchemaField( f : ClassField ){
+	static function classFieldToSchemaField( f : ClassField, typeKey:String ){
 
 		var type = {
 			pos : f.pos,
-			expr : typeToSchemaType(f.type)
+			expr : typeToSchemaType(f.type, typeKey)
 		};
-
-		var expr = macro { type : $type };
+        
+        if(typeKey.substr(0, 1) == "$")
+            typeKey = "@$__hx__" + typeKey; 
```

Comment:
I think it is necessary while sufficient to support and only support the dollar sign prefix (for now).

The origin of this whole PR is that I use a lot "type" as field name in my db. But that causes some weird problems because mongoose schema by default uses "type" to define the datatype (i.e. the typeKey). I suppose using "$type" is in general good enough for most cases as it should be rare for anyone to use "$type" as field name in their db?